src/main/java/org/apache/commons/jexl3/internal/MetaCache.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.commons.jexl3.internal;
+
+import java.lang.ref.Reference;
+import java.lang.ref.ReferenceQueue;
+import java.lang.ref.WeakReference;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.IntFunction;
+
+import org.apache.commons.jexl3.JexlCache;
+
+/**
+ * A meta-cache that tracks multiple JexlCache instances via weak references.
+ * <p>
+ *  Each JexlCache created by this MetaCache is held via a WeakReference,
+ *  allowing it to be garbage collected as soon as no strong references exist.
+ * </p>
+ * <p>
+ * This allows for collective management of multiple caches, in particular clearing all caches at once.
+ * This operation is typically called when the uberspect class loader needs to change.
+ * </p>
+ */
+public class MetaCache {
```

Comment:
This class uses an unusual 2 spaces for indentation instead of the 4 in other classes. Best to normalize to 4 spaces IMO.